src/core/features/commands/commandHooks.ts

```diff
@@ -0,0 +1,67 @@
+import { useContext, useEffect, useMemo } from 'react';
+import { isEqual } from 'lodash';
+import { WorkspaceContext } from '@features/App/Workspace';
+import { useAppSelector } from '@state/redux/hooks';
+import { useWorkspaceData } from '@state/redux/profiles/hooks';
+import { selectActiveWorkspaceInfo } from '@state/redux/profiles/profiles';
+
+import { CommandEventContext, CommandPayloads } from './CommandEventProvider';
+
+/**
+ * Returns a function that calls a command by its name
+ */
+export function useCommand() {
+	const commandEvent = useContext(CommandEventContext);
+	const { profileId } = useWorkspaceData();
+	const activeWorkspace = useAppSelector(
+		useMemo(() => selectActiveWorkspaceInfo({ profileId }), [profileId]),
+		isEqual,
+	);
+
+	return <K extends keyof CommandPayloads>(
+		commandName: K,
+		...args: CommandPayloads[K] extends void
+			? [payload?: void]
+			: [payload: CommandPayloads[K]]
+	) => {
+		if (!activeWorkspace) return;
+
+		const [payload] = args;
+		const data =
+			payload === undefined
+				? { name: commandName, workspaceId: activeWorkspace.id }
+				: {
+						name: commandName,
+						workspaceId: activeWorkspace.id,
```

Comment:
Why a command is depended on workspace? Potentially we may have global commands over workspaces